Orange/data/variable.py

```diff
@@ -266,15 +389,16 @@ def val_from_str_add(self, s):
         Convert a value from a string and adjust the number of decimals if
         `adjust_decimals` is non-zero.
         """
-        return _variable.val_from_str_add_cont(self, s)
+        if s in MISSING_VALUES:
+            return Unknown
+        self.number_of_decimals = len(s.strip().split('.', 1)[1]) if '.' in s else 0
```

Comment:
This function does not work correctly. The mechanism for adjusting the number of decimals is more complicated. Please see the original.

Also, Marko implemented the Cython code since he discovered it speeds up the file parsing considerably. After reimplementing this function correctly, please check how it affects the speed of reading large files with continuous values.